cloud/utils/retry.go

```diff
@@ -0,0 +1,29 @@
+//                           _       _
+// __      _____  __ ___   ___  __ _| |_ ___
+// \ \ /\ / / _ \/ _` \ \ / / |/ _` | __/ _ \
+//  \ V  V /  __/ (_| |\ V /| | (_| | ||  __/
+//   \_/\_/ \___|\__,_| \_/ |_|\__,_|\__\___|
+//
+//  Copyright © 2016 - 2024 Weaviate B.V. All rights reserved.
+//
+//  CONTACT: hello@weaviate.io
+//
+
+package utils
+
+import (
+	"time"
+
+	"github.com/cenkalti/backoff/v4"
+)
+
+// NewBackoff returns a Backoff that can be used to retry an operation
+// We have this function to ensure that we can use the same backoff settings in multiple places in weaviate.
+func NewBackoff() backoff.BackOff {
```

Comment:
what do you think if we pass the max retries count ? to make it configurable for the need of the requester ?

Reply:
I think if/when it's needed it would be trivial to add (With like a `WithMaxRetries` call or something) so for me i'd rather keep it minimal for now. Once we get to the step we want to tune the parameters we can adjust the code.